dali/util/nvml.h

```diff
@@ -122,15 +122,32 @@ inline void SetCPUAffinity(int core = -1) {
 
   int error = pthread_setaffinity_np(pthread_self(), sizeof(requested_set), &requested_set);
   if (error != 0) {
-      DALI_WARN("Setting affinity failed! Error code: " + to_string(error));
+    DALI_WARN("Setting affinity failed! Error code: " + to_string(error));
   }
 }
 
+
 inline void Shutdown() {
   std::lock_guard<std::mutex> lock(Mutex());
   DALI_CALL(wrapNvmlShutdown());
 }
 
+
+inline bool DoesHwDecoderExist() {
+  unsigned int device_count;
+  DALI_CALL(wrapNvmlDeviceGetCount_v2(&device_count));
+  for (unsigned int device_id = 0; device_id < device_count; device_id++) {
```

Comment:
Don't you want to have this function per device id?

Reply:
Extracted as overload